cr30/marlin/firmware-history.md

```diff
@@ -0,0 +1,190 @@
+    ```
+  - Parent [upstream firmware commit](https://github.com/MarlinFirmware/Marlin/commit/b9d9e74f2cfce8e53b315e72366dc49c5550c4c6) is `b9d9e74`
+  - Corresponding [upstream config commit](https://github.com/MarlinFirmware/Configurations/commit/0a8dfe5594ca981428c9adf76068e3cbe425c3ba) is `0a8dfe5`
+- **Diff**
+  - [Available on GitHub](https://github.com/CR30-Users/Marlin-CR30/commit/c93a97279eadbb5fb3757a7c7e7696690b90b328)
+    - Some of these changes were brought into upstream (pull requests linked above)
+    - Some were not, like the side-band multilanguage implementation
+
+### Subsequent community changes
+
+More closely evaluate [the diffs between the community and upstream](https://github.com/MarlinFirmware/Marlin/compare/2.0.x...CR30-Users:Marlin-CR30:2.0.x_cr30). Choosing mainline seems ok if the community diffs don't add improvements.
+
+- **Compare:** CR30-Users `2.0.x_cr30`
+- **Base:** Nearest-neighbor official Marlin firmware with co-temporal Offical Marlin CR-30 config files
+  - Which upstream commit is the parent?
+    ```
+    $ git log --oneline -n3
+    4c8cd975d6 (HEAD, tag: latest, tag: 2.0.9.3_cr30, origin/HEAD, origin/2.0.x_cr30, 2.0.x_cr30) Merge remote-tracking branch 'upstream/2.0.x' into 2.0.x_cr30
+    16f92d673a 🔨 Add .vscode/extensions.json
+    f17f458e91 🐛 Fix RRW Keypad & Zonestar buttons (#23388)
+    ```
+  - Parent [upstream firmware commit](https://github.com/MarlinFirmware/Marlin/commits/73b8320e9caac23873169c8e10344f2f8060b389) is `73b8320` _(just after 2.0.9.3)_
+  - Corresponding [upstream config commit](https://github.com/MarlinFirmware/Configurations/commit/820725c157fe82720ab4b1f236d885c7f2dd84ea) is `820725c`
+- **Diff**
+  - The branch comparison above reflects most of this
+    - However that comparison also shows both _release_ and _bugfix_ branches being merged from upstream, so some history is remixed with each rebase
+    - Further the comparison is against the **default** configuration files from Marlin, not against the CR-30 configuration files from Marlin
+  - We can recreate a pull request diff by taking the file trees from the upstream commits linked above and diffing the unified tree against the latest tree from CR30-Users
+    1. Base: [Copy the upstream CR-30 configuration](./history/0001-Marlin-apply-CR-30-configuration-to-default-files.patch.md) into the upstream firmware tree
+    1. Compare: [Apply the CR30-Users changes](./history/0002-CR30-Users-Add-community-customizations.patch.md) to the unified upstream tree
+  - These differences are listed below
+
+#### List of Changes made in `CR30-Users`
+
+| Keep? | Difference | Reason | Note |
+|:-----:|------------|--------|------|
+| ❌ | Disable serial port 2 | bundle with rework-required changes | 🔷 |
+| ❌ | Increase hotend max temp to 275 | seems a little dangerous |  |
+| ✅ | Add PID autotune to the LCD menu |  |  |
+| ❌ | Add `BELT_KINEMATICS_DEV` configuration option | doesn't build, code history lost |  |
```

Comment:
This and the move-C-axis additions were from Marlin experiments. In the CR-30 Discord server NAK 3D Designs explained that he never built or used the firmware with `BELT_KINEMATICS_DEV`.